milli/src/update/index_documents/transform.rs

```diff
@@ -327,8 +334,15 @@ impl<'a, 'i> Transform<'a, 'i> {
     }
 
     // Flatten a document from the fields ids map contained in self and insert the new
-    // created fields.
-    fn flatten_from_fields_ids_map(&mut self, obkv: KvReader<FieldId>) -> Result<Vec<u8>> {
+    // created fields. Returns `None` if the document doesn't need to be flattened.
+    fn flatten_from_fields_ids_map(&mut self, obkv: KvReader<FieldId>) -> Result<Option<Vec<u8>>> {
+        if obkv
+            .iter()
+            .all(|(_, value)| !json_depth_checker::should_flatten_from_unchecked_slice(value))
```

Comment:
Couldn't using `any` here you be easier to understand?

Reply:
if all fields don't need to be flattened vs if not any field needs to be flattened? :thinking: 
I think I prefer having the not inside of the closure instead of out of all the iterator chain wdyt?

```rust
        if obkv
            .iter()
            .all(|(_, value)| !json_depth_checker::should_flatten_from_unchecked_slice(value))
// VS
        if !obkv
            .iter()
            .any(|(_, value)| json_depth_checker::should_flatten_from_unchecked_slice(value))
```

Reply:
Yeah, that's probably better :)